[YAML design] Filtering for vulnerability exposure (#47114)
fleetdm/fleet: docs/Configuration/yaml-files.md

    historical_data:
      uptime: true
      vulnerabilities: false
    vulnerability_exposure_historical_reporting:
      software_filters:
        - operating_system
        - browsers
        - microsoft_office
        - adobe
      has_known_exploit: true

>       epss_score:
>         min: 0
>         max: 100
>       cvss_score:
>         min: 9
>         max: 10

Reviewer: For the purpose of keeping the YAML and API close together, can we flatten these to:

* epss_min
* epss_max
* cvss_min
* cvss_max

```suggestion
      epss_min: 0
      epss_max: 100
      cvss_min: 9
      cvss_max: 10
```

Reviewer: Also CVSS should be left out of this revision or else a comment put in that it won't do anything yet.

      host_batch_size: 0
```

> ### vulnerability_settings

Reviewer 2: ```suggestion
### vulnerability_exposure_historical_reporting
```

> The `vulnerability_settings` section lets you define filters for the **Vulnerability exposure** dashboard chart (risk registry). These filters control which software and vulnerabilities are included when calculating vulnerability exposure.

Reviewer 2: ```suggestion
The `vulnerability_exposure_historical_reporting` section lets you define filters for the **Vulnerability exposure** dashboard chart (risk registry). These filters control which software and vulnerabilities are included when calculating vulnerability exposure.
```

- `software_categories` is a list of software categories to include in the vulnerability chart. Valid values are `os` (Operating system), `browser` (Google Chrome, Safari, Mozilla Firefox, Brave, and Opera), `microsoft_office` (Word, Excel, Powerpoint, and Outlook), and `adobe` (Acrobat, Flash, and Shockwave Player). If omitted, all categories are included.
- `has_known_exploit` when `true`, only includes software that has vulnerabilities which have been actively exploited in the wild (default: `false`).
- `severity` filters vulnerabilities by CVSS v3 severity score (range 0.0 to 10.0 in 0.1 increments):
  - `min` is the minimum CVSS v3 score to include (default: `0`).
  - `max` is the maximum CVSS v3 score to include (default: `10`).
- `probability_of_exploit` filters vulnerabilities by probability of exploit score (range 0 to 100):
  - `min` is the minimum probability of exploit score to include (default: `0`).
  - `max` is the maximum probability of exploit score to include (default: `100`).
- `exclude_cves` is a list of specific CVE identifiers to exclude from the vulnerability chart.

> Can only be configured for "All fleets" (`org_settings`).

Reviewer 2: ```suggestion
Can be configured for "All fleets" (`org_settings`) and specific fleets (`settings`).
```

#### Example

```yaml
org_settings:

>   vulnerability_settings:

Reviewer 2: ```suggestion
  vulnerability_exposure_historical_reporting:
```

>     software_categories:

Reviewer 2: ```suggestion
    software_filters:
```

>       - os

Reviewer 2: ```suggestion
      - operating_system
```

>       - browser

Reviewer 2: ```suggestion
      - browsers
```

      - microsoft_office
      - adobe
    has_known_exploit: true

>     probability_of_exploit:

Reviewer 2: ```suggestion
    epss_score:
```

      min: 0
      max: 100

>     severity:

Reviewer 2: ```suggestion
    cvss_score:
```

      min: 9
      max: 10

>     exclude_cves:

Reviewer 2: ```suggestion
    exclude_vulnerabilities:
```
